src/drivers/kinetis/i2c.c

```diff
@@ -0,0 +1,310 @@
+#include <MK64F12.h>
+#include <drivers/i2c.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/* SCL clock divider table for baudrate calculation */
+static const uint16_t SCL_LUT[] = {
+    20,  22,  24,  26,   28,   30,   34,   40,   28,   32,   36,   40,   44,   48,   56,   68,
+    48,  56,  64,  72,   80,   88,   104,  128,  80,   96,   112,  128,  144,  160,  192,  240,
+    160, 192, 224, 256,  288,  320,  384,  480,  320,  384,  448,  512,  576,  640,  768,  960,
+    640, 768, 896, 1024, 1152, 1280, 1536, 1920, 1280, 1536, 1792, 2048, 2304, 2560, 3072, 3840};
+
+/* global variable to store the channel parameters */
+volatile I2C_Channel i2c_chs[I2C_NUM_DEVICES];
```

Comment:
I would suggest adding a `static` here - remember that `static` means "file private", not "constant"

Reply:
We need it outside (mentioned above) :(

Reply:
Dangerous, but I see the necessity.  I had to go and write a test to prove that you cannot have something declared as `extern static`, which is highly unfortunate.  We should remember this for the future in case it somehow manages to become an issue (it shouldn't).